internal/pkg/addon/storage.go

```diff
@@ -73,6 +89,19 @@ type DDBLocalSecondaryIndex struct {
 	Name         *string
 }
 
+// RDSProps holds RDS-specific properties for addon.NewRDS().
+type RDSProps struct {
+	*StorageProps
```

Comment:
honestly I think it's okay if we just inline `ClusterName` as a field here instead of using `*StorageProps` maybe we can refactor and even remove this type.